### PR TITLE
Hardcode the JS stream prefix 

### PR DESCRIPTION
**Pull request by VladislavPaskar**
**Description**

Changes proposed in this pull request:

- Use a hardcoded Jetstream prefix whenever an event is being published (EPP part) and received from NATS backend (eventing-controller)
- Adjust the tests, remove redundant ones

**Related issue(s)**
https://github.com/kyma-project/kyma/issues/13825

### CODE REVIEW

File: components/event-publisher-proxy/testing/fixtures.go [hunk 1]
```diff
@@ -14,6 +14,7 @@ const (
 
 	MessagingNamespace            = "/messaging.namespace"
 	MessagingEventTypePrefix      = "prefix"
+	JSStreamPrefix                = "sap.kyma.custom"
```
Comment (k15r) [inline components/event-publisher-proxy/testing/fixtures.go:17]: as `sap.kyma.custom` is already our event type prefix, would it make sense to use something different for the 
StreamPrefix? 
e.g. `kyma`, `eventing`, ...

File: components/eventing-controller/pkg/deployment/publisher_deployment.go [hunk 2]
```diff
@@ -330,10 +330,10 @@ func getNATSEnvVars(natsConfig env.NatsConfig, publisherConfig env.PublisherConf
 				},
 			},
 		},
-		// jetstream-specific config
+		// Jetstream-specific config
 		{Name: "ENABLE_JETSTREAM_BACKEND", Value: strconv.FormatBool(natsConfig.EnableJetStreamBackend)},
 		{Name: "JS_STREAM_NAME", Value: natsConfig.JSStreamName},
-		{Name: "JS_STREAM_SUBJECT_PREFIX", Value: natsConfig.JSStreamSubjectPrefix},
+		{Name: "JS_STREAM_SUBJECT_PREFIX", Value: env.JetstreamSubjectPrefix},
```
Comment (k15r) [inline components/eventing-controller/pkg/deployment/publisher_deployment.go:336]: imho this setting does not need to be modified. we could just hardcode it everywhere. No real need for a environment variable

File: components/eventing-controller/pkg/env/nats_config.go [hunk 3]
```diff
@@ -2,17 +2,12 @@ package env
...
 
 	"github.com/kelseyhightower/envconfig"
 )
 
-var (
-	// invalidSubjectPrefixCharacters used to match and replace non-alphanumeric characters in the subject-prefix
-	// as per JetStream spec https://docs.nats.io/running-a-nats-service/nats_admin/jetstream_admin/naming.
-	invalidSubjectPrefixCharacters = regexp.MustCompile("[^a-zA-Z0-9.]")
-)
+const JetstreamSubjectPrefix = "sap.kyma.custom"
```
Comment (k15r) [inline components/eventing-controller/pkg/env/nats_config.go:10]: Same point as here:
https://github.com/kyma-project/kyma/pull/14262/files#r880149151

**Review by k15r — CHANGES_REQUESTED** [review]

File: components/event-publisher-proxy/testing/fixtures.go (hunk 1 above)
Comment (VladislavPaskar) [reply components/event-publisher-proxy/testing/fixtures.go:17]: I agree, changed to `kyma`

File: components/eventing-controller/pkg/deployment/publisher_deployment.go (hunk 2 above)
Comment (VladislavPaskar) [reply components/eventing-controller/pkg/deployment/publisher_deployment.go]: done

File: components/eventing-controller/pkg/env/nats_config.go (hunk 3 above)
Comment (VladislavPaskar) [reply components/eventing-controller/pkg/env/nats_config.go]: done

File: components/eventing-controller/pkg/handlers/jetstream.go [hunk 4]
```diff
@@ -258,12 +260,9 @@ func (js *JetStream) DeleteSubscription(subscription *eventingv1alpha1.Subscript
...
 }
 
-// GetJsSubjectToSubscribe appends stream name to subject if needed.
-func (js *JetStream) GetJsSubjectToSubscribe(subject string) string {
-	if strings.HasPrefix(subject, js.config.JSStreamSubjectPrefix) {
-		return subject
-	}
-	return fmt.Sprintf("%s.%s", js.config.JSStreamSubjectPrefix, subject)
+// GetJestreamSubject appends the prefix to subject.
+func (js *JetStream) GetJestreamSubject(subject string) string {
```
[the remark this answers is not in the document]
Comment (VladislavPaskar) [reply components/eventing-controller/pkg/handlers/jetstream.go]: done

**Review by k15r — APPROVED** [review]

**Review by klaudiagrz — APPROVED** [review]

**Pull request merged**